packages/mds-transactions-service/tests/service.spec.ts

```diff
@@ -105,22 +108,67 @@ describe('Transaction Service Tests', () => {
     }
   })
 
+  function* bulkTransactions(): Generator<TransactionDomainModel> {
+    let timestamp = Date.now() - 20 * 1000
+    for (let i = 0; i < 20; i++) {
+      yield {
+        transaction_id: uuid(),
+        provider_id,
+        device_id,
+        timestamp,
+        amount: 100, // "I'd buy THAT for a dollar!"
+        fee_type: 'base_fee',
+        receipt
+      }
+      timestamp += 1000
+    }
+  }
+
+  it('Bulk Post 20 Transactions', async () => {
+    const transactions = await TransactionServiceClient.createTransactions([...bulkTransactions()])
+    expect(transactions[0].device_id).toEqual(device_id)
+    // TODO more checks
+  })
+
   it('Get All Transactions', async () => {
-    const transactions = await TransactionServiceClient.getTransactions({})
-    expect(transactions.length).toEqual(1)
+    const { transactions } = await TransactionServiceClient.getTransactions({})
+    expect(transactions.length).toEqual(10)
     const [transaction] = transactions
     expect(transaction.transaction_id).toEqual(transaction_id)
   })
 
-  it('Get All Transactions with provider serach', async () => {
-    const transactions = await TransactionServiceClient.getTransactions({ provider_id })
-    expect(transactions.length).toEqual(1)
+  it('Get All Transactions with provider search and paging', async () => {
+    const { transactions, cursor } = await TransactionServiceClient.getTransactions({ provider_id })
+    expect(transactions.length).toEqual(10) // page size
     const [transaction] = transactions
     expect(transaction.transaction_id).toEqual(transaction_id)
+    const { transactions: transactions2, cursor: cursor2 } = await TransactionServiceClient.getTransactions({
+      provider_id,
+      after: cursor.afterCursor || undefined
+      // before: cursor.beforeCursor || undefined
+    })
+    expect(transactions2.length).toEqual(10) // page siz
+    const { transactions: transactions3 } = await TransactionServiceClient.getTransactions({
+      provider_id,
+      after: cursor2.afterCursor || undefined
+      // before: cursor2.beforeCursor || undefined
+    })
+    expect(transactions3.length).toEqual(1) // page siz
```

Comment:
Just to make sure I understand, 1 isn't the page size here, it's that the # of transactions is equal to 10x + 1 and you've gotten to the last page?

Reply:
Yeah, these tests are side-effect-y and at the time this test is run, there should be 21 transactions in the DB. So you go through 3 pages at 10 entries a pop, and by the time you get to the 3rd page there should only be 1 returned (cause you've already went through 20). I'm going to rework these tests so that's more clear and there are less side-effects.